scripts/scil_generate_sampling_scheme.py

```diff
@@ -151,6 +146,11 @@ def main():
     # Compute bval list
     if args.bvals is not None:
         bvals = args.bvals
+        unique_bvals = np.unique(bvals)
+        if len(unique_bvals) != S:
+            parser.error('You have provided {} shells '.format(S) +
+                         'but only {} unique bvals.'.format(len(unique_bvals)))
```

Comment:
I would remove the word only, since you can give more bval than bvec (and then the sentence make less sense)